maestro-ios-xctest-runner/maestro-driver-iosUITests/Routes/Handlers/ViewHierarchyHandler.swift

```diff
@@ -175,8 +177,12 @@ struct ViewHierarchyHandler: HTTPHandler {
         }
     }
 
-    func elementHierarchy(xcuiElement: XCUIElement) throws -> AXElement {
+    private func elementHierarchy(xcuiElement: XCUIElement) throws -> AXElement {
         let snapshotDictionary = try xcuiElement.snapshot().dictionaryRepresentation
         return AXElement(snapshotDictionary)
     }
+    
+    enum ViewHierarchyError: Error {
```

Comment:
Better use struct for Error types, even better to put that struct in the modules/ folder and call it:
```
enum AppErrorType { case client, server }
struct AppError: Error, Codable {
    let message: String
    let errorType: AppErrorType
}
```

Then we can (in a separate PR) refactor all places from:
```
            let errorJson = """
                 { "errorMessage" : "Snapshot failure while getting view hierarchy", "errorCode": "\(errorCode)" }
                """
            return HTTPResponse(statusCode: .badRequest, body:  Data(errorJson.utf8))
```
to:
```
 throw AppError(message: "foo", errorType: .server)
```

And add a server-wide exception handler that does something like:
```
catch {
 if error is AppError {
  let body = JSONEncoder().encode(error)
  return HTTPResponse(statusCode: error.errorTypeAsHTTPCode, body:  body)
 } else {
  let appError = AppError(message: error.localizedDescription, errorType: .server)
  return HTTPResponse(statusCode: appError.errorTypeAsHTTPCode, body:  JSONEncoder().encode(appError))
 }
```

Reply:
Makes sense, I have changed accordingly and added `ViewHierarchyError`. Let me know if that looks good!